src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,264 @@
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+/**
+ * Abstract class wrapping around mutli
+ */
+public abstract class AbstractCombinedItemHandler
+    implements IItemHandler, IItemHandlerModifiable, INBTSerializable<NBTTagCompound>, IWorldNameableModifyable
+{
+
+    ///NBT Constants
+    private static final String NBT_KEY_HANDLERS = "Handlers";
+    private static final String NBT_KEY_HANDLERS_INDEXLIST = "Index";
+    private static final String NBT_KEY_NAME = "Name";
+
+
+    private final IItemHandlerModifiable[] handlers;
+
+    @Nullable
+    private String customName;
+
+    @NotNull
+    private final String defaultName;
+
+    /**
+     * Method to create a new combined item handler.
+     * @param defaultName The default name of this Handler.
+     * @param handlers The combining handlers.
+     */
+    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final IItemHandlerModifiable... handlers) {
+        this.handlers = handlers;
+        this.defaultName = defaultName;
+    }
+
+    private AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
+        this.customName = customName;
+        this.defaultName = defaultName;
+    }
+
+    /**
+     * Get the name of this object. For players this returns their username
+     */
+    @Override
+    public String getName()
+    {
+        return hasCustomName() ? customName : defaultName;
+    }
+
+    /**
+     * Method to set the name of this {@link IWorldNameable}
+     *
+     * @param name The new name of this {@link IWorldNameable}, or null to reset it to its default.
+     */
+    @Override
+    public void setName(@Nullable final String name)
+    {
+        this.customName = name;
+    }
+
+    /**
+     * Returns true if this thing is named
+     */
+    @Override
+    public boolean hasCustomName()
+    {
+        return customName != null;
+    }
+
+    /**
+     * Get the formatted ChatComponent that will be used for the sender's username in chat
+     */
+    @Override
+    public ITextComponent getDisplayName()
+    {
+        return new TextComponentString(getName());
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        final NBTTagCompound compound = new NBTTagCompound();
+
+        int index = 0;
+        final NBTTagList handlerList = new NBTTagList();
+        final NBTTagList indexList = new NBTTagList();
+        for(IItemHandlerModifiable handlerModifiable : handlers) {
+            if (handlerModifiable instanceof INBTSerializable)
+            {
+                INBTSerializable serializable = (INBTSerializable) handlerModifiable;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'serializable' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)